mmaction/core/evaluation/average_precision.py

```diff
@@ -0,0 +1,337 @@
+import os.path as osp
+import pickle
+from collections import defaultdict
+
+import numpy as np
+
+from .bbox_overlaps import iou2d, spatio_temporal_iou3d, spatio_temporal_nms3d
+
+
+def pr_to_ap(precision_recall):
+    """Compute AP given precision-recall.
+
+    Args:
+        precision_recall (np.ndarray): precision_recall is an Nx2 array with
+            first column being precision and second column being recall.
+
+    Returns:
+        np.ndarray: The result of average precision.
+    """
+
+    recall_diff = precision_recall[1:, 1] - precision_recall[:-1, 1]
+    precision_sum = precision_recall[1:, 0] + precision_recall[:-1, 0]
+
+    return np.sum(recall_diff * precision_sum * 0.5)
```

Comment:
Can `mmit_mean_average_precision`, `mean_average_precision`(these two are very much similar) use this function? I guess it's what @innerlee called duplication.

```
    for i in range(len(scores)):
        precision, recall, _ = binary_precision_recall_curve(
            scores[i], labels[i])
        ap = -np.sum(np.diff(recall) * np.array(precision)[:-1])
        results.append(ap)
```